### PR TITLE
[4.0] Inject the database into the query helper

### PR DESCRIPTION
**Pull request by laoneo**
Inject the database into the query helper.

### CODE REVIEW

File: components/com_content/Helper/QueryHelper.php [hunk 1]
```diff
@@ -166,15 +170,16 @@ public static function orderbySecondary($orderby, $orderDate = 'created')
...
 	 *
 	 * @return  string  The SQL field(s) to order by.
 	 *
 	 * @since   1.6
 	 */
-	public static function getQueryDate($orderDate)
+	public static function getQueryDate($orderDate, DatabaseInterface $db = null)
 	{
-		$db = Factory::getDbo();
+		$db = $db?: Factory::getDbo();
```
Comment (Quy) [inline components/com_content/Helper/QueryHelper.php]:
```suggestion
		$db = $db ?: Factory::getDbo();
```

**Comment by PhilETaylor** [comment]
I have not tested this item.<br /><br />applied patch, navigated around, nothing broke, not tested exhaustively <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22887">issues.joomla.org/tracker/joomla-cms/22887</a>.</sub>

**Comment by bonzani** [comment]
I have tested this item :white_check_mark: successfully on d528ce92b1a2a397b04e99219a0d2e4a95353606<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22887">issues.joomla.org/tracker/joomla-cms/22887</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**